aviatrix/resource_aviatrix_external_device_conn.go

```diff
@@ -0,0 +1,417 @@
+	}
+	backupBgpLocalAsNumber, err := strconv.Atoi(d.Get("backup_bgp_remote_as_number").(string))
+	if err == nil {
+		externalDeviceConn.BackupBgpRemoteAsNumber = backupBgpLocalAsNumber
+	}
+
+	directConnect := d.Get("direct_connect").(bool)
+	if directConnect {
+		externalDeviceConn.DirectConnect = "true"
+	}
+
+	haEnabled := d.Get("enable_ha").(bool)
+	if haEnabled {
+		externalDeviceConn.HAEnabled = "true"
+	}
+
+	backupDirectConnect := d.Get("backup_direct_connect").(bool)
+	if backupDirectConnect {
+		externalDeviceConn.BackupDirectConnect = "true"
+	}
+
+	enableEdgeSegmentation := d.Get("enable_edge_segmentation").(bool)
+	if enableEdgeSegmentation {
+		externalDeviceConn.EnableEdgeSegmentation = "true"
+	}
+
+	if externalDeviceConn.ConnType == "bgp" && externalDeviceConn.RemoteSubnet != "" {
+		return fmt.Errorf("'remote_subnet' is needed for connection type of 'static' not 'bpg'")
+	} else if externalDeviceConn.ConnType == "static" && (externalDeviceConn.BgpLocalAsNumber != 0 || externalDeviceConn.BgpRemoteAsNumber != 0) {
+		return fmt.Errorf("'bgp_local_as_number' and 'bgp_remote_as_number' are needed for connection type of 'bgp' not 'static'")
+	}
+
+	customAlgorithms := d.Get("custom_algorithms").(bool)
+	if !customAlgorithms && (externalDeviceConn.Phase1Auth != "" || externalDeviceConn.Phase1DhGroups != "" ||
+		externalDeviceConn.Phase1Encryption != "" || externalDeviceConn.Phase2Auth != "" ||
+		externalDeviceConn.Phase2DhGroups != "" || externalDeviceConn.Phase2Encryption != "") {
+		return fmt.Errorf("'custom_algorithms' is not enabled, all algorithms fields should be left empty")
+	}
+
+	if haEnabled {
+		if externalDeviceConn.BackupRemoteGatewayIP == "" {
+			return fmt.Errorf("ha is enabled, please specify 'backup_remote_gateway_ip'")
+		}
+		if externalDeviceConn.BackupBgpRemoteAsNumber == 0 && externalDeviceConn.ConnType == "bgp" {
+			return fmt.Errorf("ha is enabled, and 'conn_type' is 'bgp', please specify 'backup_bgp_remote_as_number'")
+		}
+	} else {
+		if backupDirectConnect {
+			return fmt.Errorf("ha is not enabled, please set 'back_direct_connect' to false")
+		}
+		if externalDeviceConn.BackupPreSharedKey != "" || externalDeviceConn.BackupLocalTunnelCidr != "" || externalDeviceConn.BackupRemoteTunnelCidr != "" {
+			return fmt.Errorf("ha is not enabled, please set 'backup_pre_shared_key', 'backup_local_tunnel_cidr' and 'backup_remote_tunnel_cidr' to empty")
+		}
+		if externalDeviceConn.BackupBgpRemoteAsNumber != 0 && externalDeviceConn.ConnType == "bgp" {
+			return fmt.Errorf("ha is not enabled, and 'conn_type' is 'bgp', please specify 'backup_bgp_remote_as_number' to empty")
+		}
+	}
+
+	err = client.CreateExternalDeviceConn(externalDeviceConn)
+	if err != nil {
+		return fmt.Errorf("failed to create Aviatrix external device connection: %s", err)
+	}
+
+	d.SetId(externalDeviceConn.ConnName + "~" + externalDeviceConn.VpcID)
+	return resourceAviatrixExternalDeviceConnRead(d, meta)
```

Comment:
again, require the defred read as  multiple operations are  performed above

Reply:
Here, only one operations is performed.